src/Database/LSMTree/Internal.hs

```diff
@@ -1211,6 +1211,7 @@ snapshot resolve snap label th = do
                     (RW.unsafeAcquireWriteAccess (tableContent thEnv))
                     (atomically . RW.unsafeReleaseWriteAccess (tableContent thEnv))
                     $ \reg content -> do
+        supplyCredits (Entry.unNumEntries $ case confWriteBufferAlloc conf of AllocNumEntries x -> x) (tableLevels content)
```

Comment:
When we flush the buffer here, it might be underfull, which could mess up the scheduling. The conservative approach is to supply credits as if the buffer was full, and then flush the (possibly) underfull buffer. This code should be augmented with a TODO to find more performant alternatives. However, note that this bit of code here is probably going to change anyway because of https://github.com/IntersectMBO/lsm-tree/issues/392

Reply:
Moved this comment into a TODO in the code.